src/migration/1706110501504-UserStreakTrigger.ts

```diff
@@ -0,0 +1,45 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class UserStreakTrigger1706110501504 implements MigrationInterface {
+  name = 'UserStreakTrigger1706110501504';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "user_streak" ALTER COLUMN "lastViewAt" DROP NOT NULL`,
+    );
+
+    // Create the trigger function
+    await queryRunner.query(`
+      CREATE OR REPLACE FUNCTION create_user_streak()
+      RETURNS TRIGGER AS $$
+      BEGIN
+          INSERT INTO user_streak ("userId")
```

Comment:
AFAIK, this is how `user_streak` table was created:
```
CREATE TABLE "user_streak" ("userId" character varying(36) NOT NULL, "currentStreak" integer NOT NULL DEFAULT '0', "totalStreak" integer NOT NULL DEFAULT '0', "maxStreak" integer NOT NULL DEFAULT '0', "lastViewAt" TIMESTAMP WITH TIME ZONE NOT NULL, "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(), CONSTRAINT "PK_3d8ec0aea914605bd0455034673" PRIMARY KEY ("userId"))
```

the `lastViewAt` column is `NOT NULL`, but we do not specify a default value. Here we are not inserting a value either. Is that expected to work?

Reply:
Also, the test should show how the trigger works.

Reply:
Ahh, didn't see that :) thanks!